### PR TITLE
Fix incorrect platformchannel response when clipboard getData is not supported

### PR DESCRIPTION
**Pull request by ferhatb**
Correctly returns null response when method is not supported instead of causing a PlatformException.

https://github.com/flutter/flutter/issues/80073

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [x] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

### CODE REVIEW

File: lib/web_ui/test/engine/platform_dispatcher_test.dart [hunk 1]
```diff
@@ -55,5 +60,22 @@ void testMain() {
...
+        'flutter/platform',
+        codec.encodeMethodCall(MethodCall(
+          'Clipboard.getData',
+        )),
+        completer.complete,
+      );
+      final ByteData? response = await completer.future;
+      expect(() => codec.decodeEnvelope(response!) ,
+         throwsA(isA<PlatformException>()),
+      );
```
Comment (mdebbar) [inline lib/web_ui/test/engine/platform_dispatcher_test.dart:78]: nit: formatting is off.

File: lib/web_ui/test/engine/platform_dispatcher_test.dart [hunk 2]
```diff
@@ -12,6 +14,9 @@ import 'package:test/bootstrap/browser.dart';
 import 'package:test/test.dart';
 
 void main() {
+  // Patch browser so that clipboard api is not available.
+  dynamic clipboard = js_util.getProperty(html.window.navigator, 'clipboard');
+  js_util.setProperty(html.window.navigator, 'clipboard', null);
```
Comment (mdebbar) [inline lib/web_ui/test/engine/platform_dispatcher_test.dart:19]: This is only relevant to a single test. I suggest we do this in the test.

Also, please remember to reset the value of `window.navigator.clipboard`.

File: lib/web_ui/test/engine/platform_dispatcher_test.dart [hunk 3]
```diff
@@ -55,5 +60,22 @@ void testMain() {
...
+      const MethodCodec codec = JSONMethodCodec();
+      final Completer<ByteData?> completer = Completer<ByteData?>();
+      ui.PlatformDispatcher.instance.sendPlatformMessage(
+        'flutter/platform',
+        codec.encodeMethodCall(MethodCall(
+          'Clipboard.getData',
+        )),
+        completer.complete,
+      );
+      final ByteData? response = await completer.future;
```
Comment (mdebbar) [inline lib/web_ui/test/engine/platform_dispatcher_test.dart:75]: Should we check that the `response` is null?

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/test/engine/platform_dispatcher_test.dart (hunk 2 above)
Comment (ferhatb) [reply lib/web_ui/test/engine/platform_dispatcher_test.dart]: Done.

File: lib/web_ui/test/engine/platform_dispatcher_test.dart (hunk 1 above)
Comment (ferhatb) [reply lib/web_ui/test/engine/platform_dispatcher_test.dart]: Done. dartfmt.

File: lib/web_ui/test/engine/platform_dispatcher_test.dart (hunk 3 above)
Comment (ferhatb) [reply lib/web_ui/test/engine/platform_dispatcher_test.dart:76]: Done.

**Pull request merged**